addons/account/models/account_journal_dashboard.py

```diff
@@ -276,18 +276,21 @@ def get_journal_dashboard_datas(self):
             outstanding_pay_account_balance, nb_lines_outstanding_pay_account_balance = self._get_journal_outstanding_payments_account_balance(
                 domain=[('parent_state', '=', 'posted')])
 
-            self._cr.execute('''
-                SELECT COUNT(st_line.id)
-                FROM account_bank_statement_line st_line
-                JOIN account_move st_line_move ON st_line_move.id = st_line.move_id
-                JOIN account_move_line aml ON aml.move_id = st_line_move.id
-                WHERE st_line_move.journal_id IN %s
-                AND NOT st_line.is_reconciled
-                AND st_line_move.to_check IS NOT TRUE
-                AND st_line_move.state = 'posted'
-                AND aml.account_id = %s
-            ''', [tuple(self.ids), self.default_account_id.id])
-            number_to_reconcile = self.env.cr.fetchone()[0]
+            if self.default_account_id:
+                self._cr.execute('''
+                    SELECT COUNT(st_line.id)
+                    FROM account_bank_statement_line st_line
+                    JOIN account_move st_line_move ON st_line_move.id = st_line.move_id
+                    JOIN account_move_line aml ON aml.move_id = st_line_move.id
+                    WHERE st_line_move.journal_id IN %s
+                    AND NOT st_line.is_reconciled
+                    AND st_line_move.to_check IS NOT TRUE
+                    AND st_line_move.state = 'posted'
+                    AND aml.account_id = %s
+                ''', [tuple(self.ids), self.default_account_id.id])
+                number_to_reconcile = self.env.cr.fetchone()[0]
+            else:
+                number_to_reconcile = 0
```

Comment:
```suggestion
        self._cr.execute('''
                SELECT COUNT(st_line.id)
                FROM account_bank_statement_line st_line
                JOIN account_move st_line_move ON st_line_move.id = st_line.move_id
                JOIN account_journal journal ON st_line_move.journal_id = journal.id AND journal.id in %s
                JOIN account_move_line aml ON aml.move_id = st_line_move.id AND aml.account_id = journal.default_account_id
                WHERE NOT st_line.is_reconciled
                AND st_line_move.to_check IS NOT TRUE
                AND st_line_move.state = 'posted'
        ''', [tuple(self.ids)])
        number_to_reconcile = self.env.cr.fetchone()[0]
```
Wouldn't something like this avoid the query without needing to add the if?
Although skipping it would probably not be bad for performances I guess...

Reply:
just a matter of performances